applications/launchpad/gui-react/src/components/Onboarding/OnboardingMessages/LastStepsMessages.tsx

```diff
@@ -1,33 +1,175 @@
+import { useTheme } from 'styled-components'
 
 /**
  * @TODO - how the Blockchain synchronization should to work?
  */
-const messages = [
-  () => {
-    const dispatch = useAppDispatch()
-
-    return (
-      <>
-        <Text as='span' type='defaultHeavy'>
-          {t.onboarding.lastSteps.message1} ✨💪
+
+const Progress = ({ progress, time }: { progress: number; time: string }) => {
+  const theme = useTheme()
+
+  return (
+    <ProgressContainer>
+      <Text
+        color={theme.accent}
+        type='smallMedium'
+        style={{
+          alignSelf: 'flex-start',
+          marginBottom: theme.spacingVertical(0.5),
+        }}
+      >
+        {t.onboarding.lastSteps.blockchainIsSyncing}
+      </Text>
+      <ProgressBar value={progress} />
+      <RemainingTime>
+        <Text type='smallMedium'>
+          {time} {t.common.adjectives.remaining.toLowerCase()}
         </Text>
-        <Button
-          variant='button-in-text'
-          onClick={() => dispatch(setOnboardingComplete(true))}
-        >
-          <Text as='span' type='defaultUnder'>
-            Exit data synchronization
-          </Text>
-        </Button>
-      </>
-    )
-  },
-]
-
-export default messages
+      </RemainingTime>
+    </ProgressContainer>
+  )
+}
+
+export const BlockchainSyncStep = ({
+  pushMessages,
+  updateMessageBoxSize,
+}: {
+  pushMessages: (msgs: TBotMessage[]) => void
+} & TBotMessageHOCProps) => {
+  const dispatch = useAppDispatch()
+
+  const contentRef = useRef<HTMLDivElement | null>(null)
+
+  const [error, setError] = useState(false)
+  const [syncStarted, setSyncStarted] = useState(false)
+  const [syncFinished, setSyncFinished] = useState(false)
+  const [progress, setProgress] = useState(0)
+  const [remainingTime, setRemainingTime] = useState('55 min')
+
+  const startSync = async () => {
+    setSyncStarted(true)
+    pushMessages([
+      {
+        content: <Text>{t.onboarding.lastSteps.message2}</Text>,
+        barFill: 0.875,
+        noSkip: true,
+      },
+    ])
+    try {
+      await dispatch(baseNodeActions.startNode()).unwrap()
+    } catch (e) {
+      setError(true)
+    }
+  }
+
+  useEffect(() => {
+    if (error) {
+      pushMessages([
+        {
+          content: (
+            <>
+              <Text>{t.onboarding.lastSteps.syncError}</Text>
+              <CtaButtonContainer>
+                <Button
+                  variant='secondary'
+                  onClick={() => dispatch(setOnboardingComplete(true))}
+                >
+                  {t.common.verbs.continue}
+                </Button>
+              </CtaButtonContainer>
+            </>
+          ),
+          barFill: 0.875,
+          noSkip: true,
+          wait: 200,
+        },
+      ])
+    }
+  }, [error])
+
+  useEffect(() => {
+    if (syncStarted && updateMessageBoxSize && contentRef.current) {
+      updateMessageBoxSize()
+    }
+  }, [syncStarted])
+
+  useEffect(() => {
+    if (syncFinished) {
```

Comment:
same as above -> call this logic directly instead of adding state which fires off an useEffect